File: app/src/main/java/com/cn29/aac/ui/masterdetail/vm/SimpleMasterDetailShareViewModel.java
package com.cn29.aac.ui.masterdetail.vm;

import com.cn29.aac.repo.github.GitRepoRepository;
import com.cn29.aac.repo.github.Repo;
import com.cn29.aac.repo.util.Resource;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

/**
 * Created by charlesng0209 on 25/6/2017.
 */

public class SimpleMasterDetailShareViewModel extends ViewModel {

  private GitRepoRepository gitRepoRepository;

  public SimpleMasterDetailShareViewModel(GitRepoRepository gitRepoRepository) {
    this.gitRepoRepository = gitRepoRepository;
  }

  public LiveData<Resource<Repo>> loadRepos(String owner, String name) {
    return gitRepoRepository.loadRepo(owner, name);
  }

  public LiveData<Resource<List<Repo>>> loadRepos(String owner) {
    return gitRepoRepository.loadRepos(owner);
  }
}

File: app/src/main/java/com/cn29/aac/ui/masterdetail/vm/SimpleMasterDetailShareViewModel.kt
package com.cn29.aac.ui.masterdetail.vm

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.cn29.aac.repo.github.GitRepoRepository
import com.cn29.aac.repo.github.Repo
import com.cn29.aac.util.Result

/**
 * Created by charlesng0209 on 25/6/2017.
 */
class SimpleMasterDetailShareViewModel(private val gitRepoRepository: GitRepoRepository) :
        ViewModel() {
    fun loadRepos(owner: String,
                  name: String): LiveData<Result<Repo>> =
            gitRepoRepository.loadRepo(owner, name)

    fun loadRepos(owner: String?): LiveData<Result<List<Repo>>> =
            gitRepoRepository.loadRepo(owner!!)

}